docs/api/utils/proxyWithComputed.mdx

```diff
@@ -48,16 +36,60 @@ const state2 = proxyWithComputed(
 )
 
 // if you want a computed value to derive from another computed, you must declare the dependency first:
-const state = proxyWithComputed(
-  {
-    count: 1,
-  },
+const state3 = proxyWithComputed(
+  { count: 1 },
   {
     doubled: memoize((snap) => snap.count * 2),
     quadrupled: memoize((snap) => snap.doubled * 2),
   }
 )
 ```
 
-The last use case fails to infer types in TypeScript
-[#192](https://github.com/pmndrs/valtio/issues/192).
+The last use case fails to infer types in TypeScript, see [#192](https://github.com/pmndrs/valtio/issues/192).
+
+#### Disclaimer on `proxy-memoize`
+
+Be careful not to overuse `proxy-memoize` because `proxy-memoize` and `useSnapshot` do similar optimization and double optimization may lead to less performance.
+
+The similarity is that both `proxy-memoize` and `useSnapshot` keep caches of data; the difference is that getters invoked against `snapshot`s or `useSnapshot`s are not cached and will re-invoke the getter/re-calculate the result each time they are accessed; whereas `proxy-memoize` will use the `snap` argument to calculate the getter result only once per snapshot. )
```

Comment:
Fwiw this is my understanding of things, happy to update.

Reply:
The similarity is usage tracking (= proxy-compare). caching is kind of by-product.
I'm not sure what you are trying say for the difference.

Reply:
Oh man, I see; I thought `proxy-memoize` was just some generic memoization project from NPM that the examples happened to use.

But I get it now, it is literally a sister project, also built on `proxy-compare`, that does snapshot-ish style lazy recalcs.

That is nifty; will update the docs with that learning. Let me know if the updated description is right.

...fwiw, it seems like maybe `proxyWithComputed` should learn to use "real"/native Valtio snapshots, and pass those into the computed's gets, so that it can do the lazy recalc itself? Just thinking that would solve the "double copies" problem? Granted, it would mean all computeds are cached... (unless there was an option the user passed to turn caching on/off for each getter.) Dunno, just talking out loud.